packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -540,8 +555,13 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
   }
 }
 
+/** Get the value from the given state and props (converting from number to string if needed) */
 function _getValue(props: ITextFieldProps, state: ITextFieldState): string | undefined {
   const { value = state.uncontrolledValue } = props;
+  if (typeof value === 'number') {
```

Comment:
Hmmm how did we get away with not having this conversion before....

Reply:
The conversion was already done in other places, but I thought it made more sense to move it here.